DFC.ServiceTaxonomy.GraphSync/GraphSyncers/Parts/EponymousPartGraphSyncer.cs

```diff
@@ -114,6 +120,68 @@ public class EponymousPartGraphSyncer : IContentPartGraphSyncer
             return Enumerable.Empty<ICommand>();
         }
 
+        public async Task<bool> VerifySyncComponent(ContentItem contentItem, INode node, ContentTypePartDefinition contentTypePartDefinition,
+            IEnumerable<IRelationship> relationships, IEnumerable<INode> destNodes)
+        {
+            foreach (var field in contentTypePartDefinition.PartDefinition.Fields)
+            {
+                JObject value = contentItem.Content[contentItem.ContentType][field.Name];
+
+                if (field.FieldDefinition.Name == "ContentPickerField")
+                {
+                    var relationshipType = $"ncs__has{field.Settings["ContentPickerFieldSettings"]!["DisplayedContentTypes"]![0]}";
+                    var contentItemIds = (JArray)value["ContentItemIds"]!;
+
+                    var contentCount = contentItemIds.Count;
+                    //TODO : need to ignore case for hasSocCode vs hasSOCCode - need to make sure these line up!
+                    var relationshipCount = relationships.Count(x => string.Equals(x.Type, relationshipType, StringComparison.CurrentCultureIgnoreCase));
+
+                    if (contentCount != relationshipCount)
+                    {
+                        return false;
+                    }
+
+                    foreach (var item in contentItemIds)
+                    {
+                        var contentItemId = (string)item!;
+
+                        var destContentItem = await _session
+                            .Query<ContentItem, ContentItemIndex>(x =>
+                                x.ContentItemId == contentItemId).FirstOrDefaultAsync();
+
+                        var destUri = (string)destContentItem.Content.GraphSyncPart.Text;
+
+                        var destNode = destNodes.SingleOrDefault(n => (string)n.Properties["uri"] == destUri);
+
+                        if (destNode == null)
+                        {
+                            return false;
+                        }
+
+                        var relationship = relationships.SingleOrDefault(x =>
+                            string.Equals(x.Type, relationshipType, StringComparison.CurrentCultureIgnoreCase) && x.EndNodeId == destNode.Id);
+
+                        if (relationship == null)
+                        {
+                            return false;
+                        }
+                    }
+                }
+                else
+                {
+                    var contentItemValue = value["Text"] ?? value["Html"] ?? value["Value"];
```

Comment:
We'll have to be careful we don't fail validation for a field we don't sync.
We should also check `AddSyncComponents` works correctly for fields we haven't explicitly added support for.
Probably outside the scope of this PR though, so perhaps we just add a todo for now.

Reply:
Not sure what you mean - what fields don't we sync?

Reply:
Currently we don't sync fields such as Boolean and DateTime.
We need to ensure the validation doesn't fail if a content type contains any fields we don't (yet) sync, as it's valid for them to be in the ContentItem but not the node properties.